### ISSUE TITLE
Sleep per week is sleep per 24hrs

### ISSUE THREAD

**Issue by forkoz** [Freeyourgadget/Gadgetbridge#866]
#### Before opening an issue please confirm the following:
- [x ] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [ x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x ] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

#### Your issue is:
Sleep per week does not populate for multiple days.

#### Your wearable device is:

Mi1s

#### Your android version is:
6.0.1
#### Your Gadgetbridge version is:
22.1

As you can see the date range is only 1 day on the bottom and as soon as the next day passes you lose the other days. 
![nov1-2](https://user-images.githubusercontent.com/12354546/32383405-e9d1e228-c085-11e7-9e46-efe0396403de.png)
![nov2-3](https://user-images.githubusercontent.com/12354546/32383407-eb1331be-c085-11e7-9bac-0bb305dc43d0.png)

**Comment by IzzySoft** [comment]
I'm not 100% sure as I didn't investigate deep enough to confirm the exact values (grin) – but I think I can confirm that for the Pebble Time Steel. Whenever I looked I had the same feeling. Always looked on Mondays, so I thought it might be the week-break :innocent:

**Comment by forkoz** [comment]
Yesterday WED had a bar just like THU screen shot. Tomorrow I find out for sure if the behavior continues or if I'm just crazy. In contrast the steps have a wed and thursday today and if I flip back I have just wed.

**Comment by ashimokawa** [comment]
Unfortunately I cannot reproduce these problems :/
I Guess I will have to charge and wear my mi1s

**Comment by forkoz** [comment]
It finally populated a bar this morning but I did some bad things that screw up the test.
1. I slept at 2am - 6am some time to 10am
2. Compiled my own version to get the commits from 22.2
BTW, the unmentionable app counts this at 8hrs of sleep. There are h/r measurements throughout the whole cycle. So how to count a broken sleep cycle when there is HR data?

I will check again tomorrow and then I'll close this if it keep updating unless I see something funky.
![nov3-4](https://user-images.githubusercontent.com/12354546/32406686-d95610b6-c14a-11e7-86d3-27e40f55505d.png)

**Comment by IzzySoft** [comment]
I checked mine this morning as well: weekly sleep stats showed exactly the same numbers as those for the last night. Cannot be "accidentally matching" as usually I sleep about 7-8h, but last night it was almost 10h. So something is wrong here.

Btw: the "red average line" in the bar graph (lower half of the "sleep a week" page) shows 8:20. But then I just note the "circle graph" is labeled "sleep today", so it probably is supposed to be the same as the one on the "your sleep" tab – which means I've been on the wrong track all the time …

Will keep an eye on it – but with that detail it looks at least to me it's working as it should. Normally you want to know how long you've slept "last night"; so if you went to bed before midnight, it would make no sense to make a split there and say "you've slept 6h today" if you went to bed at 9pm, right? It's just a little confusing to have the same data in the "circle graph" on both tabs – but once you've got the idea (note the last night is still missing from the bar graph then) it seems OK to me.

**Comment by ashimokawa** [comment]
@IzzySoft 
While I have problems following your conversation, I still want to note that "Your sleep" shows sleep in the last 24h while "sleep a week" shows sleep from 12:00-12:00. These can differ. For example when you get up early at 6:00 but you were sleeping till 10:00 the day before you still have that extra 4h in your 24h range while the "sleep a week" shows only the last night, because it is 12:00-12.00.

**Comment by forkoz** [comment]
Heh, that we'd never know unless we read the code. So basically you're shorted 1 day on the graphs. I.e, this is 3 days of data but I get only 2 bars.  

More stats would help here vs just the giant circle. It also rotates around for some reason.

The distinction between other app ignoring my cut in the middle sleep vs GB treating hours asleep as awake? was also interesting. The band itself recorded my heart rate like I never woke up and started early then paused. 

~11:51PM Starts recording
~12:58AM Stops Recording
then
2:12AM Starts Recording
10:04AM Stops Recording

![brokensleep](https://user-images.githubusercontent.com/12354546/32410165-3b0ec04a-c188-11e7-9f95-c69c2a45c34c.png)

**Comment by IzzySoft** [comment]
@ashimokawa Yes, I've noticed I've been on the wrong track. Thanks for your explanations (and me finally noticing the small-print next to the weekly graph), now I'm in the tracks.

> "Your sleep" shows sleep in the last 24h while "sleep a week" shows sleep from 12:00-12:00.

Then, how likely is it that both circle graphs show the exact same values at any time? At this very moment, the display matches your explanations: in "sleep a week" the graph is empty as I wasn't yet sleeping today. But this morning they showed the same values. Going back one day in history now that's no longer the case (puzzled).

OK, I'll watch that a bit closer. Please ignore the parts of my comments above which don't make sense due to my misunderstanding :wink:

**Comment by IzzySoft** [comment]
@ashimokawa is there something we should transfer to the wiki from this issue – or do we simply close it as solved?
